File: software/examples/java/ExampleCallback.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletOutdoorWeather;

public class ExampleCallback {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Outdoor Weather Bricklet
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletOutdoorWeather ow = new BrickletOutdoorWeather(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Enable station data callbacks
		ow.setStationCallbackConfiguration(true);

		// Enable sensor data callbacks
		ow.setSensorCallbackConfiguration(true);

		// Add station data listener
		ow.addStationDataListener(new BrickletOutdoorWeather.StationDataListener() {
			public void stationData(int identifier, int temperature, int humidity,
			                        long windSpeed, long gustSpeed, long rain,
			                        int windDirection, boolean batteryLow) {
				System.out.println("Identifier (Station): " + identifier);
				System.out.println("Temperature (Station): " + temperature/10.0 + " °C");
				System.out.println("Humidity (Station): " + humidity + " %RH");
				System.out.println("Wind Speed (Station): " + windSpeed/10.0 + " m/s");
				System.out.println("Gust Speed (Station): " + gustSpeed/10.0 + " m/s");
				System.out.println("Rain (Station): " + rain/10.0 + " mm");

				if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_N) {
					System.out.println("Wind Direction (Station): N");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_NNE) {
					System.out.println("Wind Direction (Station): NNE");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_NE) {
					System.out.println("Wind Direction (Station): NE");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_ENE) {
					System.out.println("Wind Direction (Station): ENE");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_E) {
					System.out.println("Wind Direction (Station): E");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_ESE) {
					System.out.println("Wind Direction (Station): ESE");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_SE) {
					System.out.println("Wind Direction (Station): SE");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_SSE) {
					System.out.println("Wind Direction (Station): SSE");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_S) {
					System.out.println("Wind Direction (Station): S");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_SSW) {
					System.out.println("Wind Direction (Station): SSW");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_SW) {
					System.out.println("Wind Direction (Station): SW");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_WSW) {
					System.out.println("Wind Direction (Station): WSW");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_W) {
					System.out.println("Wind Direction (Station): W");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_WNW) {
					System.out.println("Wind Direction (Station): WNW");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_NW) {
					System.out.println("Wind Direction (Station): NW");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_NNW) {
					System.out.println("Wind Direction (Station): NNW");
				} else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_ERROR) {
					System.out.println("Wind Direction (Station): Error");
				}

				System.out.println("Battery Low (Station): " + batteryLow);
				System.out.println("");
			}
		});

		// Add sensor data listener
		ow.addSensorDataListener(new BrickletOutdoorWeather.SensorDataListener() {
			public void sensorData(int identifier, int temperature, int humidity) {
				System.out.println("Identifier (Sensor): " + identifier);
				System.out.println("Temperature (Sensor): " + temperature/10.0 + " °C");
				System.out.println("Humidity (Sensor): " + humidity + " %RH");
				System.out.println("");
			}
		});

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Outdoor Weather Bricklet

	// Callback function for station data callback
	static void StationDataCB(BrickletOutdoorWeather sender, byte identifier,
	                          short temperature, byte humidity, long windSpeed,
	                          long gustSpeed, long rain, byte windDirection,
	                          bool batteryLow)
	{
		Console.WriteLine("Identifier (Station): " + identifier);
		Console.WriteLine("Temperature (Station): " + temperature/10.0 + " °C");
		Console.WriteLine("Humidity (Station): " + humidity + " %RH");
		Console.WriteLine("Wind Speed (Station): " + windSpeed/10.0 + " m/s");
		Console.WriteLine("Gust Speed (Station): " + gustSpeed/10.0 + " m/s");
		Console.WriteLine("Rain (Station): " + rain/10.0 + " mm");

		if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_N)
		{
			Console.WriteLine("Wind Direction (Station): N");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_NNE)
		{
			Console.WriteLine("Wind Direction (Station): NNE");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_NE)
		{
			Console.WriteLine("Wind Direction (Station): NE");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_ENE)
		{
			Console.WriteLine("Wind Direction (Station): ENE");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_E)
		{
			Console.WriteLine("Wind Direction (Station): E");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_ESE)
		{
			Console.WriteLine("Wind Direction (Station): ESE");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_SE)
		{
			Console.WriteLine("Wind Direction (Station): SE");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_SSE)
		{
			Console.WriteLine("Wind Direction (Station): SSE");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_S)
		{
			Console.WriteLine("Wind Direction (Station): S");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_SSW)
		{
			Console.WriteLine("Wind Direction (Station): SSW");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_SW)
		{
			Console.WriteLine("Wind Direction (Station): SW");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_WSW)
		{
			Console.WriteLine("Wind Direction (Station): WSW");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_W)
		{
			Console.WriteLine("Wind Direction (Station): W");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_WNW)
		{
			Console.WriteLine("Wind Direction (Station): WNW");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_NW)
		{
			Console.WriteLine("Wind Direction (Station): NW");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_NNW)
		{
			Console.WriteLine("Wind Direction (Station): NNW");
		}
		else if(windDirection == BrickletOutdoorWeather.WIND_DIRECTION_ERROR)
		{
			Console.WriteLine("Wind Direction (Station): Error");
		}

		Console.WriteLine("Battery Low (Station): " + batteryLow);
		Console.WriteLine("");
	}

	// Callback function for sensor data callback
	static void SensorDataCB(BrickletOutdoorWeather sender, byte identifier,
	                         short temperature, byte humidity)
	{
		Console.WriteLine("Identifier (Sensor): " + identifier);
		Console.WriteLine("Temperature (Sensor): " + temperature/10.0 + " °C");
		Console.WriteLine("Humidity (Sensor): " + humidity + " %RH");
		Console.WriteLine("");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletOutdoorWeather ow = new BrickletOutdoorWeather(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Enable station data callbacks
		ow.SetStationCallbackConfiguration(true);

		// Enable sensor data callbacks
		ow.SetSensorCallbackConfiguration(true);

		// Register station data callback to function StationDataCB
		ow.StationDataCallback += StationDataCB;

		// Register sensor data callback to function SensorDataCB
		ow.SensorDataCallback += SensorDataCB;

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}
